### PR TITLE
Add Sidre list format to sphinx documentation

### PR DESCRIPTION
**Pull request by nselliott**
# Summary

- This PR is documentation
- It does the following (modify list as needed):
  - Adds documentation of the Sidre list format to the sphinx docs
  - Addresses #977

### CODE REVIEW

File: src/axom/sidre/docs/sphinx/group.rst [hunk 1]
```diff
@@ -226,6 +226,55 @@ Move and Copy Views
    view data; i.e., it is shared by the original and the new view in the
    destination group.
 
+----------------------------
+List Format
+----------------------------
+
+The list format is an alternate way for a group to hold its child groups and
+view. In this format, any number of child group or view items can be created to
```
Comment (agcapps) [inline src/axom/sidre/docs/sphinx/group.rst:234]:
```suggestion
views. In this format, any number of child group or view items can be created to
```

File: src/axom/sidre/docs/sphinx/group.rst [hunk 2]
```diff
@@ -226,6 +226,55 @@ Move and Copy Views
...
+be held and accessed like a list, using iterators that iterate in the order
+that they were added to the parent group. Child groups and views held in the
+list format cannot be accessed by name.
```
Comment (agcapps) [inline src/axom/sidre/docs/sphinx/group.rst:237]:
```suggestion
views. In list format, any number of child group or view items can be created.
Each can be accessed, in order of creation, using an iterator over groups or
over views. Child groups and views held in list format cannot be accessed by
name.
```

**Comment by agcapps** [comment]
Other than the rephrasing suggestion for the first paragraph, it looks good and is very welcome.  Thanks for this.

**Review by agcapps — APPROVED** [review]

File: src/axom/sidre/docs/sphinx/group.rst [hunk 3]
```diff
@@ -226,6 +226,55 @@ Move and Copy Views
...
+
+  .. code-block:: C++
+          
+     Group* g0 = list_group->createUnnamedGroup();
+     Group* g1 = list_group->createUnnamedGroup();
+     Group* g2 = list_group->createUnnamedGroup();
+     View* v0 = list_group->createView("");
+     View* v1 = list_group->createViewScalar("", 1.0);
+     View* v2 = list_group->createViewString("", "foo");
+     View* v3 = list_group->createViewg("", type, num_elems, buffer);
```
Comment (kennyweiss) [inline src/axom/sidre/docs/sphinx/group.rst:264]: `createViewg(...)`

I think there's a type in this line. Please fix.

**Review by kennyweiss — APPROVED** [review]
Thanks @nselliott!

File: src/axom/sidre/docs/sphinx/group.rst [hunk 4]
```diff
@@ -226,6 +226,55 @@ Move and Copy Views
...
+     Group* g0 = list_group->createUnnamedGroup();
+     Group* g1 = list_group->createUnnamedGroup();
+     Group* g2 = list_group->createUnnamedGroup();
+     View* v0 = list_group->createView("");
+     View* v1 = list_group->createViewScalar("", 1.0);
+     View* v2 = list_group->createViewString("", "foo");
+     View* v3 = list_group->createView("", type, num_elems, buffer);
+
+While it is allowed to pass a non-empty string to be the name of a child
+item held in the list format, a string with path syntax, (such as
```
Comment (rhornung67) [inline src/axom/sidre/docs/sphinx/group.rst:267]:
```suggestion
item held in the list format, a string with path syntax such as
```

File: src/axom/sidre/docs/sphinx/group.rst [hunk 5]
```diff
@@ -226,6 +226,55 @@ Move and Copy Views
...
+``"foo/bar/baz"``) will be considered invalid, and the object creation methods
```
Comment (rhornung67) [inline src/axom/sidre/docs/sphinx/group.rst:268]:
```suggestion
``"foo/bar/baz"`` will be considered invalid, and the object creation methods
```

File: src/axom/sidre/docs/sphinx/group.rst [hunk 6]
```diff
@@ -226,6 +226,55 @@ Move and Copy Views
...
+will return a null pointer if such a string is provided.
```
Comment (rhornung67) [inline src/axom/sidre/docs/sphinx/group.rst:269]: Here it is "null pointer" and below (line 275) it is "nullptr". They should be consistent.

**Review by rhornung67 — APPROVED** [review]
Thanks! @nselliott

**Pull request merged**